instrumentation/jmx-metrics/javaagent/src/test/java/io/opentelemetry/instrumentation/javaagent/jmx/JmxMetricInsightInstallerTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.javaagent.jmx;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.instrumentation.jmx.yaml.JmxConfig;
+import io.opentelemetry.instrumentation.jmx.yaml.JmxRule;
+import io.opentelemetry.instrumentation.jmx.yaml.RuleParser;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+
+class JmxMetricInsightInstallerTest {
+  private static final String PATH_TO_ALL_EXISTING_RULES = "src/main/resources/jmx/rules";
+  private static final Set<String> FILES_TO_BE_TESTED =
+      new HashSet<>(
+          Arrays.asList(
+              "activemq.yaml",
+              "hadoop.yaml",
+              "jetty.yaml",
+              "kafka-broker.yaml",
+              "tomcat.yaml",
+              "wildfly.yaml"));
+
+  @Test
+  void testToVerifyExistingRulesAreValid() throws Exception {
+    File existingRulesDir = new File(PATH_TO_ALL_EXISTING_RULES);
+    assertThat(existingRulesDir == null).isFalse();
+
+    RuleParser parser = RuleParser.get();
+    assertThat(parser == null).isFalse();
+
+    // make sure we have correct number of files
+    File[] existingRules = existingRulesDir.listFiles();
+    assertThat(existingRules).hasSize(FILES_TO_BE_TESTED.size());
```

Comment:
On mac directories often contain a hidden `.DS_Store` file. I think it is better to modify the next loop so that it remembers which of the files it saw and after the loop verify whether it saw all the files that you expect.